### PR TITLE
Split funcs into separate submodules

### PR DESCRIPTION
**Pull request by yanchith**
This PR almost doesn't change any code, only restructures the `interpreter_funcs` module into submodules. The only changed code is:

- Rename `FuncImpl*` -> `Func*`
- The `operations` module was removed and the implementations contained there (transform and shrinkwrap) were inlined into their respective funcs.

### CODE REVIEW

File: src/interpreter_funcs/mod.rs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+/// Returns the global set of function definitions available to the
+/// editor.
+///
+/// Note that since funcs can have internal state such as a cache or
+/// random state, two instances of the function table are not always
+/// equivalent.
+pub fn create_function_table() -> BTreeMap<FuncIdent, Box<dyn Func>> {
+    let mut funcs: BTreeMap<FuncIdent, Box<dyn Func>> = BTreeMap::new();
+
+    // Special funcs
```
Comment (janper) [inline src/interpreter_funcs/mod.rs]: Not sure if this should be called *special*. What about *manipulation* for now?

**Review by janper — DISMISSED** [review]
Yes, this is cleaner.
I'm wondering if you are planning to group the functions in the UI as well.

**Comment by yanchith** [comment]
> I'm wondering if you are planning to group the functions in the UI as well.

I started thinking about it. There's certainly something we can do. Let's discuss the design of it first.

File: src/interpreter_funcs/mod.rs (hunk 1 above)
Comment (yanchith) [reply src/interpreter_funcs/mod.rs]: Sure. All of it is up for discussion.

**Pull request merged**